### PR TITLE
Check for Subsequence

### PR DESCRIPTION
**Pull request by samyukta20**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Check for Subsequence

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#993

### CODE REVIEW

**Comment by samyukta20** [comment]
Kindly review this PR @iamrajiv @Kajol-Kumari @SKAUL05 @siddharth25pandey

File: C-Plus-Plus/README.md [hunk 1]
```diff
@@ -209,6 +209,8 @@ _add list here_
 - [String to int(atoi)](cp/string_to_int.cpp)
 - [Container with most water](Container_with_most_water.cpp)
 - [Spiral-Matrix](cp/Spiral_Matrix.cpp)
+- [Check for Subsequence](cp/Check_for_Subsequence.cpp)
+
```
Comment (iamrajiv) [inline C-Plus-Plus/README.md]: @samyukta20 remove this extra line break

File: C-Plus-Plus/cp/Check_for_Subsequence.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	    else
+	    {
+	        cout<<"No, str1 is not substring of str2";
+	    }
+	    cout<<endl;
+	}
+	return 0;
+}
+
+//Time Complexity: O(n)
```
Comment (iamrajiv) [inline C-Plus-Plus/cp/Check_for_Subsequence.cpp]: @samyukta20 Convert all single-line comments into one multiline comment. Something like this
```
/* 
...
*/
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@samyukta20 Format your code using code formatter.

File: C-Plus-Plus/cp/Check_for_Subsequence.cpp (hunk 2 above)
Comment (samyukta20) [reply C-Plus-Plus/cp/Check_for_Subsequence.cpp]: Changes done. Kindly check. @iamrajiv

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@samyukta20 Format your code something like this
```cpp
#include <iostream>
using namespace std;

int main() {
	int t;
	cin >> t;
	while (t--) {
		int i = 0, j = 0;                                  //i will point str1 and j will point str2
		string str1, str2;                                 //str1 is substring
		cin >> str1 >> str2;                               //str2 is full string
		while (i < str1.length() && j < str2.length()) {
			if (str1[i] == str2[j]) {
				i++;
			}
			j++;
		}
		if (i == str1.length()) {
			cout << "Yes, str1 is substring of str2";
		} else {
			cout << "No, str1 is not substring of str2";
		}
		cout << endl;
	}
	return 0;
}

/*Time Complexity: O(n)
Input:
2
WGA AWRGNA
QBR EQVBA
Output:
Yes, str1 is substring of str2
No, str1 is not substring of str2
*/
```

**Comment by samyukta20** [comment]
> @samyukta20 Format your code something like this
> 
> ```c++
> #include <iostream>
> using namespace std;
> 
> int main() {
> 	int t;
> 	cin >> t;
> 	while (t--) {
> 		int i = 0, j = 0;                                  //i will point str1 and j will point str2
> 		string str1, str2;                                 //str1 is substring
> 		cin >> str1 >> str2;                               //str2 is full string
> 		while (i < str1.length() && j < str2.length()) {
> 			if (str1[i] == str2[j]) {
> 				i++;
> 			}
> 			j++;
> 		}
> 		if (i == str1.length()) {
> 			cout << "Yes, str1 is substring of str2";
> 		} else {
> 			cout << "No, str1 is not substring of str2";
> 		}
> 		cout << endl;
> 	}
> 	return 0;
> }
> 
> /*Time Complexity: O(n)
> Input:
> 2
> WGA AWRGNA
> QBR EQVBA
> Output:
> Yes, str1 is substring of str2
> No, str1 is not substring of str2
> */
> ```

Okay

**Comment by samyukta20** [comment]
Please check now. @iamrajiv

**Review by iamrajiv — APPROVED** [review]
@samyukta20 LGTM!

File: C-Plus-Plus/cp/Check_for_Subsequence.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+/*Time Complexity: O(n)
+Input:
+2
+WGA AWRGNA
+QBR EQVBA
+Output:
+Yes, str1 is substring of str2
+No, str1 is not substring of str2
+*/
```
Comment (Kajol-Kumari) [inline C-Plus-Plus/cp/Check_for_Subsequence.cpp]: please leave one blank line at the end of the file.

**Review by Kajol-Kumari — APPROVED** [review]
@samyukta20 minor changes required, rest LGTM :+1:

File: C-Plus-Plus/cp/Check_for_Subsequence.cpp (hunk 3 above)
Comment (samyukta20) [reply C-Plus-Plus/cp/Check_for_Subsequence.cpp]: Done @Kajol-Kumari

**Comment by samyukta20** [comment]
Kindly review this PR @SKAUL05

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Comment by iamrajiv** [comment]
@all-contributors please add @salonishah01 for code

**Comment by samyukta20** [comment]
@iamrajiv please tag me to add as a contributor. Kindly check the tag above.

**Comment by iamrajiv** [comment]
@all-contributors please add @samyukta20 for code

**Pull request merged**